### DISCUSSION TITLE
Local no longer works and passwords are missing in browser extension after facing vaultwarden public through Cloudflare

### DISCUSSION THREAD

**Discussion by machielvdb** [dani-garcia/vaultwarden#3034] [Q&A] [answered]
Hi

I've been running vaultwarden for about a year now on my Raspberry Pi 4, behind a WireGuard VPN. I never used the web vault because I couldn't be bothered enabling HTTPS, the extensions worked just fine.

I recently decided to face the server public through Cloudflare, to provide easy access without VPN for a few close friends and family members wanting to use a vault. After adding my DNS Record and restarting the container to allow signups, I am able to connect to the webvault using the domain name. However, when I login into the Windows and browser extensions, using the custom server url setting does not work, I get an 'Uknown username or password' error. 

The same error happens when trying to connect through the extension on the local ip, as I did for over a year. When I add the https URL to the Web Vault URL under custom environment, I am able to login but a TON of my passwords are missing, which are showing up on the webvault.

Syncing works fine, newly added passwords through the extension disappear after a container restart. I've already placed my most recent backup, but no change. The web vault over the domain works fine and has all my credentials (over 50), while the extension has 10, all of the earliest ones I've ever added.

I use a very basic compose file:

version: '3.3'
services:
    server:
        container_name: vaultwarden
        restart: always
        environment:
            - SIGNUPS_ALLOWED=false
            - INVITATIONS_ALLOWED=false
            - ADMIN_TOKEN=<SuperToken>
        volumes:
            - '/mnt/bitwarden/data/:/data/'
        ports:
            - '9080:80'
        image: 'vaultwarden/server:latest'

I am at a loss here, if someone could give me some insight in what's happening and how to resolve it it'd greatly appreciate it.

Thanks in advance

**Comment by karkazzul** [comment, marked as answer]
Hi,

In the extension and on the desktop mobile, you have this icone: 
![image](https://user-images.githubusercontent.com/23294229/209291326-550b5ae0-5411-4633-9a90-49da04670c5d.png)
and you can enter your DNS entry for vaultwarden like: 
![image](https://user-images.githubusercontent.com/23294229/209291522-d321968e-0c90-486a-9355-447f5990f128.png)

**Reply by machielvdb** [reply]
Hi

Yes, this is how it worked for over a year. With the local ip address and port filled in the Server URL. Now that I use DNS, I seem to have connection but it refuses to let me login, both with the domain and the local ip:port. When I put the domain under 'Custom Environment - Web Vault Server URL' I am able to login but a lot of my passwords are missing through the extension.

**Reply by BlackDex** [reply]
Sounds like either the newer versions of the extensions and your Vaultwarden version are not matching. Try to use the `testing` tagged images. Or, CloudFlare is doing something with the response?

**Reply by machielvdb** [reply]
Hi BlackDex

You were right! The extensions got updated 2 days ago to 2022.12. I pulled the testing image and everything is now working as expected.

Thank you so much!!
